### PR TITLE
Sergkanz/exc stat improvements

### PR DESCRIPTION
**Pull request by SergeyKanzhelev**
Couple fixes for exc stats

### CODE REVIEW

File: Src/WindowsServer/WindowsServer.Shared/FirstChanceExceptionStatisticsTelemetryModule.cs [hunk 1]
```diff
@@ -220,13 +220,22 @@ private void TrackStatistics(string type, string operation, string method, bool
 
         private string GetDimCappedString(string value, ConcurrentDictionary<string, int> capValues)
```
Comment (cijothomas) [inline Src/WindowsServer/WindowsServer.Shared/FirstChanceExceptionStatisticsTelemetryModule.cs]: can you rename value to something like dimensionValue or something? value seem like a keyword...

File: Src/WindowsServer/WindowsServer.Shared/FirstChanceExceptionStatisticsTelemetryModule.cs [hunk 2]
```diff
@@ -220,13 +220,22 @@ private void TrackStatistics(string type, string operation, string method, bool
...
         {
-            if (capValues.Count > 100)
+            int tmp;
```
Comment (cijothomas) [inline Src/WindowsServer/WindowsServer.Shared/FirstChanceExceptionStatisticsTelemetryModule.cs]: did the unit test catch this issue? If not, consider adding a test.

Comment (dnduffy) [inline Src/WindowsServer/WindowsServer.Shared/FirstChanceExceptionStatisticsTelemetryModule.cs]: nitpick "temp"

File: Src/WindowsServer/WindowsServer.Shared/FirstChanceExceptionStatisticsTelemetryModule.cs [hunk 3]
```diff
@@ -220,13 +220,22 @@ private void TrackStatistics(string type, string operation, string method, bool
...
+            var result = capValues.TryGetValue(value, out tmp);
```
Comment (dnduffy) [inline Src/WindowsServer/WindowsServer.Shared/FirstChanceExceptionStatisticsTelemetryModule.cs]: Is this supposed to be "bool"?

File: Src/WindowsServer/WindowsServer.Shared/FirstChanceExceptionStatisticsTelemetryModule.cs [hunk 4]
```diff
@@ -220,13 +220,22 @@ private void TrackStatistics(string type, string operation, string method, bool
...
+
+            if (result)
+            {
+                return value;
+            }
+            else if (capValues.Count > 100)
```
Comment (Dmitry-Matveev) [inline Src/WindowsServer/WindowsServer.Shared/FirstChanceExceptionStatisticsTelemetryModule.cs]: If it's a concurrent code, the capValues.Count > 100 might return true and false for the same value:
- Thread A enters the method with value "ABC", capValues.Count = 100; "ABC" is not in the dictionary
- Thread B enters the method with value "ABC", capValues.Count = 100; "ABC" is not in the dictionary
- Whoever adds it first (say "A") might create a case where both "A" and "B" passed TryGetValue() and it returned nothing. Then "A" added it and capValues.Count is now 101, so "B" does not pass the check and gets "OtherValue" instead. Is this a positive concurrency on purpose or this code is not executed in parallel?

Comment (SergeyKanzhelev) [inline Src/WindowsServer/WindowsServer.Shared/FirstChanceExceptionStatisticsTelemetryModule.cs]: we do not need strong capping, just preventing from hitting ourselves in a foot. And I also wait for proper built-in functionality of dim capping in Metric Aggregation

**Pull request merged**